resources/type-system/inference.md

```diff
@@ -680,6 +686,38 @@ The constraint solution for a type variable `X` with respect to a constraint set
 Note that the constraint solution is a type schema, and hence may contain
 occurences of the unknown type.
 
+#### Constraint solution for a set of type variables
+
+The constraint solution for a set of type variables `{X0, ..., Xn}` with respect
```

Comment:
I think I know what you mean, but you haven't quite set up what it means to be a previous solution here, as far as I can tell.  That is, what does it mean to be a "previous solution"?  I initially thought that you were carrying this across the implicit iteration in this definition, but I don't think that's right - this is an input to the algorithm here, right?  If so, I think maybe "partial solution" is a better term to make this clear.  That is, I think you are setting this up so that there can be an initial set of "frozen" things that are already fixed (this could maybe use commentary to make that clear if so).   So maybe something like:

"to a constraint set C and partial solution ..."

Reply:
That's much clearer, thanks!